### PR TITLE
Feature - update IPMI documentation

### PR DESCRIPTION
**Pull request by paraenggu**
Support (and fixes) for generating IPMI based template documentation with the help of the existing `update-app-doc.xsl`. This finally allows us to share the IPMI templates on [Zabbix shares](https://share.zabbix.com/).

### CODE REVIEW

File: ipmi/Sensor_Discovery/Template_IPMI_Sensor_Discovery.dummy.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    <groups>
+        <group>
+            <name>IPMI templates</name>
+        </group>
+    </groups>
+    <templates>
+        <template>
+            <template>Template IPMI Sensor Discovery</template>
+            <name>Template IPMI Sensor Discovery</name>
+            <description>This is just an empty IPMI Sensor discovery dummy template, required for documentation generation. There's no need to install it.</description>
```
Comment (hairmare) [inline ipmi/Sensor_Discovery/Template_IPMI_Sensor_Discovery.dummy.xml]: I feel like a dummy template is a way to hacky solution for this. Why didn't you change the generator script to do it's thing without needing to render this dummy template?

File: update-app-doc.xsl [hunk 2]
```diff
@@ -285,9 +285,20 @@
   </xsl:template>
 
   <!-- toplevel boilerplate -->
-  <xsl:template match="zabbix_export"># Zabbix <xsl:value-of select="$appName"/> monitoring<xsl:text>
+  <xsl:template match="zabbix_export">
```
Comment (hairmare) [inline update-app-doc.xsl]: Did you commit the results of this change somewhere or am I missing something? Didn't we agree on getting rid of these underscores at the source?

File: ipmi/Sensor_Discovery/Template_IPMI_Sensor_Discovery.dummy.xml (hunk 1 above)
Comment (paraenggu) [reply ipmi/Sensor_Discovery/Template_IPMI_Sensor_Discovery.dummy.xml]: The problem is, that the current `xsltproc` based documentation generation needs an input XML to do it's work. Currently, `IPMI Sensor Discovery` is the only package without a corresponding Template XML and we probably won't have significantly more of them in the future.
However, if you have an alternative proposal which doesn't requires too much adaptions of the generation process, I will be more than happy to get rid of the dummy template.

File: update-app-doc.xsl (hunk 2 above)
Comment (paraenggu) [reply update-app-doc.xsl]: > Did you commit the results of this change somewhere or am I missing something?
- https://github.com/radiorabe/rabe-zabbix/pull/29/commits/edf9f4310121aa5c0f68c1126e5525f17b13273e#diff-a9e5e544b49e4b25ef08dbab2ea4916dR1
- https://github.com/radiorabe/rabe-zabbix/pull/29/commits/edf9f4310121aa5c0f68c1126e5525f17b13273e#diff-f66954e77df16b3208191299ba9a93eeR1


> Didn't we agree on getting rid of these underscores at the source?

Yes, we did, but forgot to re-address them within the `Makefile`. The `Makefile` takes the names from the directories which have and underscore in their names ([`appDir="app/${appName// /_}"`](https://github.com/radiorabe/rabe-zabbix/blob/master/README.md#adding-an-app-template), that's why we need to replace them again.

File: ipmi/Sensor_Discovery/Template_IPMI_Sensor_Discovery.dummy.xml (hunk 1 above)
Comment (paraenggu) [reply ipmi/Sensor_Discovery/Template_IPMI_Sensor_Discovery.dummy.xml]: Thanks for the proposal, addressed with 41f62301e328242e9dbdcd0ed07664277dcf0496 and 9f5823da07cc3bee23f0c3b9fe7e30eac59123d4.

**Comment by paraenggu** [comment]
@hairmare: Can we get this merged, or are there some more concerns? Thanks!

**Pull request merged**